File: dotnetcore/secrets/src/Business/SecretHelper.cs
using System.IO;

namespace FP.DockerTraining.Secrets.Business
{
public class SecretHelper
    {
        private const string defaultPath = "/run/secrets/";

        public static string GetSecret(string fileName)
        {
            var filePath = Path.Combine(defaultPath, fileName);

#if DEBUG
            filePath = Path.Combine(Directory.GetCurrentDirectory(), fileName);
#endif

            if (File.Exists(filePath))
            {
                return File.ReadAllText(filePath);
            }
            return string.Empty;
        }
    }
}

File: php/secrets/src/secrethelper.php
<?php

class SecretHelper
{
    protected static $defaultpath = '/run/secrets/';

    public static function GetSecret($secretname)
    {
        $filename = self::$defaultpath . $secretname;
        if(file_exists($filename))
        {
            return file_get_contents($filename, true);
        }
        return '';
    }
}
?>
